tasks/collector.py

```diff
@@ -36,6 +59,86 @@
 }
 
 
+class YAMLValidationError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+
+
+def find_matching_components(manifest, components_to_match: dict, present: bool) -> list:
+    """Given a manifest and dict of components to match, if present=True, return list of
+    components found, otherwise return list of components missing."""
+    res = []
+    for component_type, components in components_to_match.items():
+        for component in components:
+            found_component = False
+            components_matching_component_type = manifest.get(component_type)
+            if components_matching_component_type:
+                for module in components_matching_component_type:
+                    if module.get("gomod").find(component) != -1:
+                        found_component = True
+                        if not present:
+                            found_component = True
+                        else:
+                            res.append(component)
+                        break
+            if not present and not found_component:
+                res.append(component)
+    return res
+
+
+def validate_manifest(manifest) -> list:  # return list of components to remove, empty list if valid
+    # validate collector version matches ocb version
+    manifest_version = manifest.get("dist").get("otelcol_version")
+    if manifest_version and manifest_version != OCB_VERSION:
+        raise YAMLValidationError(
+            f"Collector version ({manifest_version}) in manifest does not match required OCB version ({OCB_VERSION})"
+        )
+
+    # validate component versions matches ocb version
+    module_types = ["extensions", "exporters", "processors", "receivers", "connectors"]
+    for module_type in module_types:
+        components = manifest.get(module_type)
+        if components:
+            for component in components:
+                for module in component.values():
+                    if module.find(OCB_VERSION) == -1:
+                        raise YAMLValidationError(
+                            f"Extension {module}) in manifest does not match required OCB version ({OCB_VERSION})"
```

Comment:
This message talks about extensions but is inside a loop iterating over all the module types. Should it say 'Component' instead of 'Extension'

Reply:
yes, changed. thank you.